examples/conditional-send/conditional_send.bal

```diff
@@ -0,0 +1,36 @@
+import ballerina/io;
+
+public function main() {
+    boolean foo = true;
+
+    worker w1 {
+        // send action can be used in a conditional context.
+        if foo {
+            10 -> w2;
+        }
+    }
+
+    worker w2 returns int|error:NoMessage {
+        int|error:NoMessage a = <- w1;
+        return a;
+    }
+
+    worker w3 {
+        if foo {
+            1 -> w4;
+        } else {
+            0 -> w4;
+        }
+    }
+
+    worker w4 returns int|error:NoMessage {
+        int|error:NoMessage d = <- w3 | w3;
```

Comment:
Should we mention how there should be an alternate receive for each conditional send?